### PR TITLE
cds: use "IsActiveEntity" in anno expressions

### PR DESCRIPTION
**Pull request by stewsk**

### CODE REVIEW

File: cds/cdl.md [hunk 1]
```diff
@@ -1105,6 +1105,7 @@ Each path in the expression is checked:
 * If the annotation is assigned to a subelement of a structured element, the top level
   elements of the entity can be accessed via `$self`.
 * A parameter `par` can be accessed via `:par`, just like parameters of a parametrized entity in queries.
+* For a draft-enabled entity, element `IsActiveEntity` can be accessed via prefix `$draft`.
```
Comment (smahati) [inline cds/cdl.md:1108]: would using the word 'property' instead of 'element' here be better?

Comment (stewsk) [reply cds/cdl.md:1108]: In CAP we usually use "element" (or maybe "field") rather than "property", so I wouldn't change it.

**Review by hjboth — APPROVED** [review]

**Review by bugwelle — APPROVED** [review]

File: cds/cdl.md [hunk 2]
```diff
@@ -1105,8 +1105,10 @@ Each path in the expression is checked:
 * If the annotation is assigned to a subelement of a structured element, the top level
   elements of the entity can be accessed via `$self`.
 * A parameter `par` can be accessed via `:par`, just like parameters of a parametrized entity in queries.
-* For a draft-enabled entity, element `IsActiveEntity` can be accessed via prefix `$draft`.
-* If a path cannot be resolved successfully, compilation fails with an error.
+* The draft specific element `IsActiveEntity` can be referred to with the magic variable `$draft.IsActiveEntity`.
```
Comment (stewsk) [inline cds/cdl.md:1108]: hm, compared to the other entries in this list, this is a rather long and detailed description for a special case

Comment (hjboth) [reply cds/cdl.md:1108]: yes that's right, it's a special case that (what I think) needs some more detailed explanation (all others in this list are straight forward and will always work)

Comment (stewsk) [reply cds/cdl.md:1108]: ok

File: cds/cdl.md [hunk 3]
```diff
@@ -1105,7 +1105,10 @@ Each path in the expression is checked:
 * If the annotation is assigned to a subelement of a structured element, the top level
   elements of the entity can be accessed via `$self`.
 * A parameter `par` can be accessed via `:par`, just like parameters of a parametrized entity in queries.
-* If a path cannot be resolved successfully, compilation fails with an error.
+* The draft specific element `IsActiveEntity` can be referred to with the magic variable `$draft.IsActiveEntity`.
+  During draft augmentation `$draft.IsActiveEntity` is rewritten to `$self.IsActiveEntity` for all draft enabled
```
Comment (smahati) [inline cds/cdl.md:1109]:
```suggestion
  During draft augmentation `$draft.IsActiveEntity` is rewritten to `$self.IsActiveEntity` for all root and sub nodes of draft enabled
```
does this rewrite mean that it's $draft.IAE in some handlers and then it's always $self.IAE?

File: cds/cdl.md [hunk 4]
```diff
@@ -1105,7 +1105,10 @@ Each path in the expression is checked:
...
+  entities (root and sub nodes but not for named types or entity parameters).
```
Comment (smahati) [inline cds/cdl.md:1110]:
```suggestion
  entities (but not for named types or entity parameters).
```

Comment (hjboth) [reply cds/cdl.md:1110]: sorry, no in a draft context the state elements are available in both the root and the sub nodes, that's important to know.

File: cds/cdl.md (hunk 3 above)
Comment (hjboth) [reply cds/cdl.md:1109]: it has nothing to do with handlers...

File: cds/cdl.md (hunk 4 above)
Comment (smahati) [reply cds/cdl.md:1110]: I mentioned that in the previous line.

Comment (hjboth) [reply cds/cdl.md:1110]: so what's the difference? In fact the content in parentheses are more like an additional information... Just leave it as it is, it's good enough

File: cds/cdl.md (hunk 3 above)
Comment (smahati) [reply cds/cdl.md:1109]:
```suggestion
 It can be reffered to by `$self.IsActiveEntity` in all root and sub nodes of draft enabled
```

Comment (hjboth) [reply cds/cdl.md:1109]: no please not. `$self.IsActiveEntity` IS A REFERENCE, a reference can't be referenced.... Why is there always the urge to improve something that is already exactly what it should be?

**Review by smahati — APPROVED** [review]

**Pull request merged**